src/meltano/core/meltano_file.py

```diff
@@ -47,6 +50,69 @@ def __init__(
             jobs=self.load_job_tasks(jobs or []),
         )
 
+        self.expansion_env = dict(os.environ)
+        self.base_plugin_environments = defaultdict(dict)
```

Comment:
We can't add new `self` properties on `Canonical` subclasses because they'll be dumped into `meltano.yml`! `meltano init` now generates:

```
version: 1
default_environment: dev
send_anonymous_usage_stats: false
environments:
- name: dev
- name: staging
- name: prod
expansion_env:
  GREP_COLOR: 37;45
  LDFLAGS: -I/opt/homebrew/opt/openssl/include -L/opt/homebrew/opt/openssl/lib
  MANPATH: '/opt/homebrew/share/man::'
  LESS_TERMCAP_mb: "\e[01;31m"
  NODENV_SHELL: zsh
  TERM_PROGRAM: iTerm.app
  LESS_TERMCAP_md: "\e[01;31m"
  PYENV_ROOT: /Users/douwemaan/.pyenv
  LESS_TERMCAP_me: "\e[0m"
  TERM: xterm-256color
  SHELL: /opt/homebrew/bin/zsh
  HOMEBREW_REPOSITORY: /opt/homebrew
  TMPDIR: /var/folders/v7/hly9rv_125d4txw9d91d8x9w0000gn/T/
  TERM_PROGRAM_VERSION: 3.4.15
  LESS_TERMCAP_ue: "\e[0m"
  TERM_SESSION_ID: w1t1p0:C65A128D-A2DC-42E5-833A-6136A6430970
  PYENV_VERSION: meltano-dev
  USER: douwemaan
  GREP_COLORS: mt=37;45
  COMMAND_MODE: unix2003
  SSH_AUTH_SOCK: /private/tmp/com.apple.launchd.hmSTNnRda1/Listeners
  PYENV_DIR: /Users/douwemaan/Development/meltano-projects
  __CF_USER_TEXT_ENCODING: 0x1F5:0x0:0x0
  PYENV_VIRTUALENV_INIT: '1'
  VIRTUAL_ENV: /Users/douwemaan/.pyenv/versions/3.9.11/envs/meltano-dev
  VIRTUAL_ENV_DISABLE_PROMPT: '12'
  PAGER: less
  PYENV_VIRTUAL_ENV: /Users/douwemaan/.pyenv/versions/3.9.11/envs/meltano-dev
  WORKON_HOME: /Users/douwemaan/.virtualenvs
  LSCOLORS: exfxcxdxbxGxDxabagacad
  LESS_TERMCAP_us: "\e[01;32m"
  PATH: /Users/douwemaan/.pyenv/versions/meltano-dev/bin:/opt/homebrew/Cellar/pyenv/2.2.5/libexec:/opt/homebrew/Cellar/pyenv/2.2.5/plugins/python-build/bin:/Users/douwemaan/.rbenv/shims:/Users/douwemaan/.nodenv/shims:/opt/homebrew/Cellar/pyenv-virtualenv/1.1.5/shims:/Users/douwemaan/.rbenv/shims:/Users/douwemaan/.nodenv/shims:/opt/homebrew/Cellar/pyenv-virtualenv/1.1.5/shims:/Users/douwemaan/.pyenv/shims:/opt/homebrew/bin:/opt/homebrew/sbin:/usr/local/bin:/usr/local/sbin:/usr/bin:/bin:/usr/sbin:/sbin:/usr/local/MacGPG2/bin
  __CFBundleIdentifier: com.googlecode.iterm2
  PWD: /Users/douwemaan/Development/meltano-projects
  EDITOR: nano
  LANG: en_US.UTF-8
  MELTANO_DISABLE_TRACKING: 'True'
  ITERM_PROFILE: Default
  PYENV_HOOK_PATH: /Users/douwemaan/.pyenv/pyenv.d:/opt/homebrew/Cellar/pyenv/2.2.5/pyenv.d:/opt/homebrew/etc/pyenv.d:/etc/pyenv.d:/usr/lib/pyenv/hooks
  XPC_FLAGS: '0x0'
  RBENV_SHELL: zsh
  XPC_SERVICE_NAME: '0'
  PYENV_SHELL: zsh
  SHLVL: '1'
  HOME: /Users/douwemaan
  COLORFGBG: 15;0
  LC_TERMINAL_VERSION: 3.4.15
  HOMEBREW_PREFIX: /opt/homebrew
  ITERM_SESSION_ID: w1t1p0:C65A128D-A2DC-42E5-833A-6136A6430970
  LESS: -g -i -M -R -S -w -X -z-4
  LOGNAME: douwemaan
  VISUAL: nano
  LESS_TERMCAP_so: "\e[00;47;30m"
  LC_CTYPE: en_US.utf-8
  BROWSER: open
  TAP_GITLAB_PRIVATE_TOKEN: glpat-KrhKXdEdiwhbyTnBKfHx
  INFOPATH: '/opt/homebrew/share/info:'
  HOMEBREW_CELLAR: /opt/homebrew/Cellar
  LC_TERMINAL: iTerm2
  LESS_TERMCAP_se: "\e[0m"
  COLORTERM: truecolor
  MELTANO_JOB_TRIGGER: cli
```

Reply:
@DouweM good callout. I've converted this back to draft while I move these expansions into the appropriate `SettingsService`s instead.

Reply:
@cjohnhanson Happy to assist on this when you get back to this part of the PR after 2.0. Could be useful to jump on a call and talk through the settings architecture and where it would make most sense to add this new logic.